Update Racon 1.5.0 to include minimap2

fraser-combe opened the pull request:
This is a small update that includes Minimap2 in the Racon Dockerfile. Allows for alignment and polishing within the same container.

Changes:

Added Minimap2 installation to the Racon Dockerfile.
Updated README with usage examples and version details.
Added 1.5.0-minimap2 to the version list in documentation.

Successful build
```bash
docker build -t racon-minimap2:1.5.0 . --no-cache                                                                    
[+] Building 179.3s (19/19) FINISHED                                                                                                                                                     docker:default 
 => [internal] load build definition from Dockerfile                                                                                                                                               0.0s 
 => => transferring dockerfile: 3.01kB                                                                                                                                                             0.0s 
 => [internal] load metadata for docker.io/library/ubuntu:jammy                                                                                                                                    0.2s 
 => [internal] load .dockerignore                                                                                                                                                                  0.0s
 => => transferring context: 2B                                                                                                                                                                    0.0s
 => CACHED [builder 1/6] FROM docker.io/library/ubuntu:jammy@sha256:0e5e4a57c2499249aafc3b40fcd541e9a456aab7296681a3994d631587203f97                                                               0.0s
 => [builder 2/6] RUN apt-get update && apt-get install --no-install-recommends -y   perl   default-jre   gnuplot   libgomp1   maven   git   wget   python3   build-essential   cmake   zlib1g-d  55.7s
 => [app 2/6] RUN apt-get update && apt-get install -y   procps   wget   python3 &&   apt-get autoclean && rm -rf /var/lib/apt/lists/*                                                            15.0s
 => [builder 3/6] RUN wget https://github.com/lbcb-sci/racon/archive/refs/tags/1.5.0.tar.gz &&   tar -xvf 1.5.0.tar.gz &&   cd racon-1.5.0 &&   mkdir build &&   cd build &&   cmake -DCMAKE_BUI  34.9s 
 => [builder 4/6] RUN racon_test                                                                                                                                                                  52.2s 
 => [builder 5/6] RUN curl -L https://github.com/lh3/minimap2/releases/download/v2.28/minimap2-2.28_x64-linux.tar.bz2 | tar -jxvf - --no-same-owner &&     mv minimap2-2.28_x64-linux /usr/local/  1.1s 
 => [builder 6/6] RUN minimap2 --version                                                                                                                                                           0.2s 
 => [app 3/6] COPY --from=builder /racon-1.5.0/build/bin/* /usr/local/bin/                                                                                                                         0.0s 
 => [app 4/6] COPY --from=builder /usr/local/minimap2/* /usr/local/bin/                                                                                                                            0.0s 
 => [app 5/6] RUN mkdir /data                                                                                                                                                                      0.2s 
 => [app 6/6] WORKDIR /data                                                                                                                                                                        0.0s 
 => [test 1/4] RUN racon --help && racon --version && minimap2 --version                                                                                                                           0.2s 
 => [test 2/4] COPY --from=builder /racon-1.5.0/test/data/* /test/                                                                                                                                 0.0s
 => [test 3/4] WORKDIR /test                                                                                                                                                                       0.0s
 => [test 4/4] RUN wget -q https://raw.githubusercontent.com/nf-core/test-datasets/modules/data/genomics/prokaryotes/bacteroides_fragilis/genome/genome.paf &&   wget -q https://github.com/nf-c  33.2s
 => exporting to image                                                                                                                                                                             0.4s
 => => exporting layers                                                                                                                                                                            0.4s
 => => writing image sha256:6d2c85b80101199fc595c63e17672568ddf0801440a76f789f683e7e1e8d26a6                                                                                                       0.0s
 => => naming to docker.io/library/racon-minimap2:1.5.0               
```
Container success for minimap2 and Racon
```bash
docker run -it --rm racon-minimap2:1.5.0 bash
root@aa678884fc3a:/test# ls /usr/local/bin
FAQ.md  LICENSE.txt  NEWS.md  README-js.md  README.md  cookbook.md  k8  minimap2  minimap2.1  paftools.js  racon  racon_test
root@aa678884fc3a:/test# ls /test
genome.fna.gz  sample_ava_overlaps.mhap.gz  sample_layout.fasta.gz  sample_overlaps.sam.gz  sample_reads.fastq.gz      test.fastq.gz  test_polished.fasta
genome.paf     sample_ava_overlaps.paf.gz   sample_overlaps.paf.gz  sample_reads.fasta.gz   sample_reference.fasta.gz  test.paf
root@aa678884fc3a:/test# exit
exit

docker run --rm racon-minimap2:1.5.0 racon --version
1.5.0

docker run --rm racon-minimap2:1.5.0 minimap2 --version
2.28-r1209
```
Pull Request (PR) checklist:
- [X] Include a description of what is in this pull request in this message.
- [X] The dockerfile successfully builds to a test target for the user creating the PR. (i.e. `docker build --tag samtools:1.15test --target test docker-builds/samtools/1.15` )
- [X] Directory structure as name of the tool in lower case with special characters removed with a subdirectory of the version number (i.e. `spades/3.12.0/Dockerfile`)
   - [ ] (optional) All test files are located in same directory as the Dockerfile (i.e. `shigatyper/2.0.1/test.sh`)
- [X] Create a simple container-specific [README.md](https://github.com/StaPH-B/docker-builds/blob/master/.github/workflow-templates/readme-template.md) in the same directory as the Dockerfile (i.e. `spades/3.12.0/README.md`)
   - [ ] If this README is longer than 30 lines, there is an explanation as to why more detail was needed
- [X] Dockerfile includes the recommended [LABELS](https://github.com/StaPH-B/docker-builds/blob/master/dockerfile-template/Dockerfile#L8-L18) 
- [X] Main [README.md](https://github.com/StaPH-B/docker-builds/blob/master/README.md) has been updated to include the tool and/or version of the dockerfile(s) in this PR
- [X] [Program_Licenses.md](https://github.com/StaPH-B/docker-builds/blob/master/Program_Licenses.md) contains the tool(s) used in this PR and has been updated for any missing

erinyoung commented on racon/1.5.0-minimap2/Dockerfile:
```diff
@@ -0,0 +1,104 @@
+WORKDIR /test
+
+# Example polishing workflow
+RUN wget -q https://raw.githubusercontent.com/nf-core/test-datasets/modules/data/genomics/prokaryotes/bacteroides_fragilis/genome/genome.paf && \
+  wget -q https://github.com/nf-core/test-datasets/raw/modules/data/genomics/prokaryotes/bacteroides_fragilis/genome/genome.fna.gz && \
+  wget -q https://github.com/nf-core/test-datasets/raw/modules/data/genomics/prokaryotes/bacteroides_fragilis/nanopore/fastq/test.fastq.gz && \
+  minimap2 -x map-ont genome.fna.gz test.fastq.gz > test.paf && \
+  racon -t 2 test.fastq.gz test.paf genome.fna.gz > test_polished.fasta && \
+  head test_polished.fasta
```
Could you add in a minimap2 test? Just to make sure that it's working.

erinyoung commented on racon/1.5.0/README.md:
```diff
@@ -1,23 +1,46 @@
 
-Code repository: https://github.com/lbcb-sci/racon
+## Tools Included
+- **Racon**: Polishes long-read assemblies.
+- **Minimap2**: Generates alignments for use with Racon and other polishing tools.
 
-Basic information on how to use this tool:
-- executable: racon
-- help: -h
-- version: -v
-- description: Polishes long read assemblies
+### Code Repositories
+- Racon: [GitHub Repository](https://github.com/lbcb-sci/racon)
```
I think this file is in the wrong directory. Should this be in the 1.5.0-minimap2 subdirectory?

fraser-combe replied:
you are correct updated and reset the README for 1.5.0 to its correct README

erinyoung commented:
Also, can you add a 'software.version' LABEL?

fraser-combe commented:
Added software version label for minimap2, minimap2 test and corrected README file for 1.5.0 back to its original version and added the 1.5.0-minimap2 README , 
Container successfully builds - output of minimap2 test
```bash
   wget -q https://raw.githubusercontent.com/lh3/minimap2/master/test/MT-human.fa && \                                                         
    wget -q https://raw.githubusercontent.com/lh3/minimap2/master/test/MT-orang.fa && \
    minimap2 -a MT-human.fa MT-orang.fa > test.sam && \
    head test.sam
"
[M::mm_idx_gen::0.002*1.51] collected minimizers
[M::mm_idx_gen::0.004*1.27] sorted minimizers
[M::main::0.004*1.26] loaded/built the index for 1 target sequence(s)
[M::mm_mapopt_update::0.004*1.25] mid_occ = 10
[M::mm_idx_stat] kmer size: 15; skip: 10; is_hpc: 0; #seq: 1
[M::mm_idx_stat::0.004*1.24] distinct minimizers: 3111 (100.00% are singletons); average occurrences: 1.000; average spacing: 5.326; total length: 16569
[M::worker_pipeline::0.012*1.06] mapped 1 sequences
[M::main] Version: 2.28-r1209
[M::main] CMD: minimap2 -a MT-human.fa MT-orang.fa
[M::main] Real time: 0.012 sec; CPU: 0.013 sec; Peak RSS: 0.005 GB
@HD     VN:1.6  SO:unsorted     GO:query
@SQ     SN:MT_human     LN:16569
@PG     ID:minimap2     PN:minimap2     VN:2.28-r1209   CL:minimap2 -a MT-human.fa MT-orang.fa
MT_orang        0       MT_human        577     60      14M2D4M3I37M1D85M1D232M1I559M1D6M1I550M1D2M1D146M2I3M1D3M1D132M1D3M1I40M3I13M1D1M1D335M3I4M1D3M2D342M1D52M1I13M3I1M2D52M1I592M1D3M1I485M1D5M1I974M3I4M3D230M1D59M1D156M1D31M1I98M1I26M14I329M3I7M3D1203M1D4M1I70M1D345M1D9M1I398M7I8M8I1M1I9M3I2M1I2M1D390M1I5M1D193M1I6M1D195M1D7M1I1826M1D10M1I1256M1D49M1D157M3D5M3I48M2I1M1I3M3D1203M1I2M2D1M1I44M2D2M1I2M1I38M2D16M2I2079M1I5M1D50M1D3M1I43M5I57M1I54M4D19M1I39M2D8M1I
```

erinyoung commented:
And the other one (there's a lot of nucleotide sequence I decided to NOT copy over)
```
#20 [test 5/5] RUN wget -q https://raw.githubusercontent.com/nf-core/test-datasets/modules/data/genomics/prokaryotes/bacteroides_fragilis/genome/genome.paf &&   wget -q https://github.com/nf-core/test-datasets/raw/modules/data/genomics/prokaryotes/bacteroides_fragilis/genome/genome.fna.gz &&   wget -q https://github.com/nf-core/test-datasets/raw/modules/data/genomics/prokaryotes/bacteroides_fragilis/nanopore/fastq/test.fastq.gz &&   minimap2 -x map-ont genome.fna.gz test.fastq.gz > test.paf &&   racon -t 2 test.fastq.gz test.paf genome.fna.gz > test_polished.fasta &&   head test_polished.fasta
#20 2.016 [M::mm_idx_gen::0.141*0.98] collected minimizers
#20 2.052 [M::mm_idx_gen::0.177*1.32] sorted minimizers
#20 2.052 [M::main::0.177*1.32] loaded/built the index for 2 target sequence(s)
#20 2.064 [M::mm_mapopt_update::0.189*1.30] mid_occ = 10
#20 2.064 [M::mm_idx_stat] kmer size: 15; skip: 10; is_hpc: 0; #seq: 2
#20 2.074 [M::mm_idx_stat::0.199*1.29] distinct minimizers: 960607 (98.44% are singletons); average occurrences: 1.022; average spacing: 5.346; total length: 5248938
#20 2.969 [M::worker_pipeline::1.094*2.14] mapped 1000 sequences
#20 2.978 [M::main] Version: 2.28-r1209
#20 2.978 [M::main] CMD: minimap2 -x map-ont genome.fna.gz test.fastq.gz
#20 2.978 [M::main] Real time: 1.103 sec; CPU: 2.346 sec; Peak RSS: 0.099 GB
#20 3.026 [racon::Polisher::initialize] loaded target sequences 0.040913 s
#20 3.410 [racon::Polisher::initialize] loaded sequences 0.383137 s
#20 3.411 [racon::Polisher::initialize] loaded overlaps 0.001046 s
#20 5.586 [racon::Polisher::initialize] aligning overlaps [=>                  ] 2.175326 s
[racon::Polisher::initialize] aligning overlaps [==>                 ] 2.942659 s
[racon::Polisher::initialize] aligning overlaps [===>                ] 3.949796 s
[racon::Polisher::initialize] aligning overlaps [====>               ] 4.505672 s
[racon::Polisher::initialize] aligning overlaps [=====>              ] 5.939849 s
[racon::Polisher::initialize] aligning overlaps [======>             ] 6.066440 s
[racon::Polisher::initialize] aligning overlaps [=======>            ] 6.318355 s
[racon::Polisher::initialize] aligning overlaps [========>           ] 6.929153 s
[racon::Polisher::initialize] aligning overlaps [=========>          ] 7.542024 s
[racon::Polisher::initialize] aligning overlaps [==========>         ] 8.327258 s
[racon::Polisher::initialize] aligning overlaps [===========>        ] 10.550879 s
[racon::Polisher::initialize] aligning overlaps [============>       ] 11.214224 s
[racon::Polisher::initialize] aligning overlaps [=============>      ] 11.763926 s
[racon::Polisher::initialize] aligning overlaps [==============>     ] 12.964983 s
[racon::Polisher::initialize] aligning overlaps [===============>    ] 13.139547 s
[racon::Polisher::initialize] aligning overlaps [================>   ] 14.274693 s
[racon::Polisher::initialize] aligning overlaps [=================>  ] 15.047569 s
[racon::Polisher::initialize] aligning overlaps [==================> ] 15.680032 s
[racon::Polisher::initialize] aligning overlaps [===================>] 18.223266 s
[racon::Polisher::initialize] aligning overlaps [====================] 18.592780 s
#20 22.02 [racon::Polisher::initialize] transformed data into windows 0.016810 s
#20 22.03 [racon::Polisher::polish] generating consensus [=>                  ] 0.007186 s
[racon::Polisher::polish] generating consensus [==>                 ] 0.007924 s
[racon::Polisher::polish] generating consensus [===>                ] 0.008159 s
[racon::Polisher::polish] generating consensus [====>               ] 0.009247 s
[racon::Polisher::polish] generating consensus [=====>              ] 0.009785 s
[racon::Polisher::polish] generating consensus [======>             ] 0.010282 s
[racon::Polisher::polish] generating consensus [=======>            ] 0.010915 s
[racon::Polisher::polish] generating consensus [========>           ] 0.014060 s
[racon::Polisher::polish] generating consensus [=========>          ] 0.014503 s
[racon::Polisher::polish] generating consensus [==========>         ] 1.451959 s
[racon::Polisher::polish] generating consensus [===========>        ] 4.280060 s
[racon::Polisher::polish] generating consensus [============>       ] 7.007702 s
[racon::Polisher::polish] generating consensus [=============>      ] 7.693067 s
[racon::Polisher::polish] generating consensus [==============>     ] 7.693341 s
[racon::Polisher::polish] generating consensus [===============>    ] 7.709327 s
[racon::Polisher::polish] generating consensus [================>   ] 7.713621 s
[racon::Polisher::polish] generating consensus [=================>  ] 7.931063 s
[racon::Polisher::polish] generating consensus [==================> ] 7.934605 s
[racon::Polisher::polish] generating consensus [===================>] 7.934935 s
[racon::Polisher::polish] generating consensus [====================] 7.935908 s
#20 29.96 [racon::Polisher::] total = 26.976569 s
#20 29.97 >NZ_CP069563.1 LN:i:5161674 RC:i:468 XC:f:0.164628
#20 29.97 ATAAATGATTGATAGTTGTGTAACCTTCGTCTCATGTTCCCATTGAATATAGG
```

erinyoung reviewed (approved):
This looks great!

I'm going to approve this PR.

What tag were you thinking about for this image? `1.5.0-minimap2`?

fraser-combe commented:
Sorry for the late reply I was away. Yes i think 1.5.0-minimap2 is a suitable tag, and thank you for the PR review!

erinyoung commented:
Perfect!

I've merged this PR and am deploying the racon:1.5.0-minimap2 image now. You can check the status here : https://github.com/StaPH-B/docker-builds/actions/runs/12244410172

Let us know if you run into any issues!
